### ISSUE TITLE
feature: Add Support for Context from 3rd Party Resources

### ISSUE THREAD

**Issue by Tiroshi** [yetone/avante.nvim#526]
### Feature request

I’m looking for a way to dynamically provide context from third-party resources (e.g., Notion pages, RAG systems) to enhance the LLM’s prompt. This context would be external to project files, such as documentation scattered across different platforms

### Motivation

In many projects, essential context is often stored outside of code files (e.g., Notion, Discord chats). The lack of integration leads to inefficiency when trying to incorporate relevant external information into the LLM prompt. An interface that allows integrating 3rd party resources directly into the prompt would solve this issue.

Current workaround: Manual copy-pasting of context, which is inefficient and disrupts the workflow.

### Other

I’d be happy to help implement this feature with your guidance. I’ve started reviewing the codebase, particularly the Ask function. Any pointers on where or how to approach this would be appreciated, and I’ll give it a go from there

**Comment by aarnphm** [comment]
extension from #372 let me implement the first few iterations, and then I can get to this.

At the end of the day, we will still try to keep it minimal within the scope of neovim.

**Comment by andnig** [comment]
In cursor one can link a documentation by simply providing a http(s) link. Cursor than crawls this page and adds it to context. Would this be a reasonable first step?

**Comment by andnig** [comment]
Yes exactly, but wouldn't this be a first huge step here? As many external ressources are just in plain web? Documentation, etc?

**Comment by luxus** [comment]
i would love to add for example the svelte 5 documentation.. will that not extend the request size? and would claude caching help here?

**Comment by aarnphm** [comment]
> Yes exactly, but wouldn't this be a first huge step here? As many external ressources are just in plain web? Documentation, etc?

that's what I'm working on see #372 lmao
